multiraft/multiraft.go

```diff
@@ -903,13 +909,17 @@ func (s *state) sendMessage(groupID uint64, msg raftpb.Message) {
 	snapStatus := raft.SnapshotFinish
 	if err != nil {
 		log.Warningf("node %v failed to send message to %v: %s", s.nodeID, nodeID, err)
-		s.multiNode.ReportUnreachable(msg.To, groupID)
+		if groupID != noGroup {
+			s.multiNode.ReportUnreachable(msg.To, groupID)
+		}
 		snapStatus = raft.SnapshotFailure
 	}
 	if msg.Type == raftpb.MsgSnap {
 		// TODO(bdarnell): add an ack for snapshots and don't report status until
 		// ack, error, or timeout.
-		s.multiNode.ReportSnapshot(msg.To, groupID, snapStatus)
+		if groupID != noGroup {
```

Comment:
You should never see a MsgSnap with no group, so this should probably panic instead of silently skipping the ReportSnapshot.